app/src/main/java/com/amaze/filemanager/ui/fragments/FtpServerFragment.kt

```diff
@@ -0,0 +1,724 @@
+import com.afollestad.materialdialogs.folderselector.FolderChooserDialog
+import com.amaze.filemanager.R
+import com.amaze.filemanager.application.AppConfig
+import com.amaze.filemanager.asynchronous.services.ftp.FtpService
+import com.amaze.filemanager.asynchronous.services.ftp.FtpService.Companion.getLocalInetAddress
+import com.amaze.filemanager.asynchronous.services.ftp.FtpService.Companion.isConnectedToLocalNetwork
+import com.amaze.filemanager.asynchronous.services.ftp.FtpService.Companion.isConnectedToWifi
+import com.amaze.filemanager.asynchronous.services.ftp.FtpService.Companion.isEnabledWifiHotspot
+import com.amaze.filemanager.asynchronous.services.ftp.FtpService.Companion.isRunning
+import com.amaze.filemanager.asynchronous.services.ftp.FtpService.FtpReceiverActions
+import com.amaze.filemanager.databinding.DialogFtpLoginBinding
+import com.amaze.filemanager.databinding.FragmentFtpBinding
+import com.amaze.filemanager.filesystem.files.CryptUtil
+import com.amaze.filemanager.ui.activities.MainActivity
+import com.amaze.filemanager.ui.notifications.FtpNotification
+import com.amaze.filemanager.ui.theme.AppTheme
+import com.amaze.filemanager.utils.OneCharacterCharSequence
+import com.amaze.filemanager.utils.Utils
+import com.google.android.material.snackbar.BaseTransientBottomBar
+import com.google.android.material.snackbar.Snackbar
+import org.greenrobot.eventbus.EventBus
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+import java.io.IOException
+import java.security.GeneralSecurityException
+import java.util.*
+
+/**
+ * Created by yashwanthreddyg on 10-06-2016. Edited by Luca D'Amico (Luca91) on 25 Jul 2017 (Fixed
+ * FTP Server while usi
+ */
+@Suppress("TooManyFunctions")
+class FtpServerFragment : Fragment(R.layout.fragment_ftp) {
+
+    private val statusText: TextView get() = binding.textViewFtpStatus
+    private val url: TextView get() = binding.textViewFtpUrl
+    private val username: TextView get() = binding.textViewFtpUsername
+    private val password: TextView get() = binding.textViewFtpPassword
+    private val port: TextView get() = binding.textViewFtpPort
+    private val sharedPath: TextView get() = binding.textViewFtpPath
+    private val ftpBtn: Button get() = binding.startStopButton
+    private val ftpPasswordVisibleButton: ImageButton get() = binding.ftpPasswordVisible
+    private var accentColor = 0
+    private var spannedStatusNoConnection: Spanned? = null
+    private var spannedStatusConnected: Spanned? = null
+    private var spannedStatusUrl: Spanned? = null
+    private var spannedStatusSecure: Spanned? = null
+    private var spannedStatusNotRunning: Spanned? = null
+    private var snackbar: Snackbar? = null
+
+    private var _binding: FragmentFtpBinding? = null
+    private val binding get() = _binding!!
+
+    private val mainActivity: MainActivity get() = requireActivity() as MainActivity
+
+    @Suppress("LabeledExpression")
```

Comment:
What is this supression?

Reply:
Seems this is the only way to get out of the lambda if the result of `StartActivityForResult` below doesn't have an `Uri` attached, as Android Studio suggested so.